test/metabase/api/card_test.clj

```diff
@@ -3457,9 +3439,10 @@
   This function exists to deduplicate test logic for all API endpoints that must return `based_on_upload`,
   including GET /api/collection/:id/items and GET /api/card/:id"
   [request]
-  (mt/with-driver :h2 ; just test on H2 because failure should be independent of drivers
-    (mt/with-temporary-setting-values [uploads-enabled true]
+  (mt/test-driver :h2 ; just test on H2 because failure should be independent of drivers
+    (mt/with-discard-model-updates [:model/Database] ; to restore any existing metabase_database.uploads_enabled=true
```

Comment:
do we really need this? AFAIK this test only modifies the db that it creates, which will be cleaned up after.